activerecord/lib/active_record/relation/where_clause.rb

```diff
@@ -107,9 +111,12 @@ def equality_node?(node)
         end
 
         def non_conflicting_binds(other)
-          conflicts = referenced_columns & other.referenced_columns
-          conflicts.map! { |node| node.name.to_s }
-          binds.reject { |attr| conflicts.include?(attr.name) }
+          # Choose all predicates that had bound params to work with them based on index
+          predicates_with_binds = predicates.select { |predicate| equality_node?(predicate) && predicate.right.is_a?(Arel::Nodes::BindParam) }
```

Comment:
@matthewd I've updated the code to do a traversal to find all the bind params so that we can keep track of the indexes, as well as updated the test to cover the problem. Let me know what you think. Thanks